kerastuner/tuners/hyperband.py

```diff
@@ -11,29 +11,43 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-import copy
+import math
 import random
 
 from ..engine import multi_execution_tuner
 from ..engine import oracle as oracle_module
-from ..engine import trial as trial_lib
 
 
 class HyperbandOracle(oracle_module.Oracle):
     """Oracle class for Hyperband.
 
+    Note that to use this Oracle with your own subclassed Tuner, your Tuner
+    class must understand three special hyperparameters that will optionally
+    be set by this Tuner:
+
+      - tuner/trial_id: The trial_id of the Trial to load from when starting
+          this trial.
+      - tuner/initial_epoch: The initial epoch the Trial should be started
+          from.
+      - tuner/epochs: The cumulative number of epochs this Trial should be
+          trained.
+
+    These hyperparameters will be set during the "successive halving" portion
+    of the Hyperband algorithm.
+
     Attributes:
         objective: String or `kerastuner.Objective`. If a string,
           the direction of the optimization (min or max) will be
           inferred.
-        max_trials: Int. Total number of trials
-            (model configurations) to test at most.
-            Note that the oracle may interrupt the search
-            before `max_trial` models have been tested.
+        max_sweeps: Int >= 1. The number of times to iterate over the full
+          Hyperband algorithm. This corresponds to a grid search over possible
+          B/nu ratios. One sweep will run approximately
```

Comment:
B and nu were not previously defined. Either define them beforehand in a detailed explanation of the algo, or don't mention them.

If the docstring does not contain sufficient information to fully understand the algo (which may be ok if it's too complex), then it should give the user actionable information to make good configuration choices (in a black box kind of way). When would you want max_sweeps=1? When would you want max_sweeps=10? What are the implications? etc.

Reply:
Done, removed. Changed the arg name to "hyperband_iterations" so that the user does not have to know what a "sweep" means

Defaulted to 1 but added a note in the docstring saying it is recommended to set this as high as your resource budget allows